src/index.tsx

```diff
@@ -1044,15 +1044,17 @@ export function useFieldInitialValue<Value>(
   const field = useAtomValue(fieldAtom, options);
   const store = useStore(options);
 
-  React.useEffect(() => {
-    if (initialValue === undefined) {
-      return;
-    }
-    if (!store.get(field.dirty)) {
-      store.set(field.value, initialValue);
-    }
+  if (initialValue === undefined) {
+    return;
+  }
+
+  if (!store.get(field.dirty) && initialValue !== store.get(field.value)) {
```

Comment:
I've just realized that for my list the value is array, which is never equal to the initial value, as it's a derived atom.
https://github.com/form-atoms/field/blob/3183dfd4ffda80e9d225582952dfb2be795509b7/src/atoms/list-atom/listAtom.ts#L187

maybe callback like `isFieldInitialized(initialValue, field, store)` can used

so I can supply my own, and either compare array deeply, or keep the last initialValue in a separate atom on the list, or other strategy...